### ISSUE TITLE
Learning rate decay

### ISSUE THREAD

**Issue by pefi9** [torch/optim#74]
In the function optim.sgd the learning rate decay is implemented this way:
line 71: local clr = lr / (1 + nevals*lrd)
Where nevals is equal to state.evalCounter. However, this evalCounter is on the line 85 increased any time when optim method is called during the training. i.e. each mini-batch update. So, the nevals not contains number of iteration, but number of mini-batches.
Is that a bug or your intention?

Thanks,
Petr

**Comment by koraykv** [comment]
I am not sure what is the difference between number of iterations and number of mini-batches in your question but in any case, the code reflects the intended behavior.
